### PR TITLE
AUTOM-2880 New style guide StatsCard component.

### PR DESCRIPTION
**Pull request by alexabreu**
image

image

## Merge Checklist

- [x] Includes link to associated Github Issue or JIRA ticket
https://jira.sendgrid.net/browse/AUTOM-2880
- [x] Has descriptive PR title and body
- [x] All changes are backwards compatible
- [x] QA'd by the SendGrid Style Guide designer
- [x] Code reviewed by a SendGrid frontend lead
- [x] If adding a new component, the component has been added to the exports in `src/index.ts`.

### CODE REVIEW

File: src/stats-card/StatsCard.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  },
+  {
+    label: 'spam reports',
+    value: '0.3%',
+  },
+];
+
+const statsWithTypes: Array<CardStatType> = [
+  {
+    label: 'emails sent',
```
Comment (chrischalstrom) [inline src/stats-card/StatsCard.stories.tsx]: type missing?

File: src/stats-card/StatsCard.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+stories.add('StatsCard with stats prop', () => (
+  <Fragment>
+    <StatsCard stats={statsWithoutTypes} />
+    <br />
+    <StatsCard stats={statsWithTypes} />
+  </Fragment>
+));
+
+stories.add('StatsCard via composition', () => (
```
Comment (chrischalstrom) [inline src/stats-card/StatsCard.stories.tsx]: new stories need new imageshots?

File: src/stats-card/CardStat.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      className,
+      value,
+      label,
+      type,
+      children,
+      ...attributes
+    } = this.props;
+
+    return (
+      <div className={cn(Styles['card-stat'], className)} {...attributes}>
```
Comment (chrischalstrom) [inline src/stats-card/CardStat.tsx]: I think our plan is to deprecate the css module way of adding classnames in ui components and just expect styleguide to be present and loaded via an external stylesheet.  Also we may want to apply custom styles to (or make assertions on) these elements from our apps.  So maybe it makes sense to add plaintext classnames mapping to the class names included in the css modules?  That way we can target the classes in a user friendly way in cypress or for our own custom rules in our apps

File: src/stats-card/StatsCard.stories.tsx (hunk 1 above)
Comment (alexabreu) [reply src/stats-card/StatsCard.stories.tsx]: it's not a required prop. type just sets the css color, and it must match one of a few set strings.

File: src/stats-card/StatsCard.stories.tsx (hunk 2 above)
Comment (alexabreu) [reply src/stats-card/StatsCard.stories.tsx]: just pushed

File: src/stats-card/CardStat.tsx (hunk 3 above)
Comment (alexabreu) [reply src/stats-card/CardStat.tsx]: fixed

File: src/stats-card/StatsCard.module.scss [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+// .email-card-stats has been genericized as .stats-card.
+// This class should be deprecated after markup changes have been made to UIComponents.
+.stats-card,
+.email-card-stats {
+  display: flex;
+  flex-grow: 1;
+  justify-content: space-around;
+  width: 100%;
+  padding: 30px;
```
Comment (richbachman) [inline src/stats-card/StatsCard.module.scss]:
```suggestion
  padding: $scale-base-up-06;
```

File: src/stats-card/StatsCard.module.scss [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+  .email-stat {
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    text-align: center;
+  }
+
+  .label {
+    margin-left: 0;
+    font-size: 12px;
```
Comment (richbachman) [inline src/stats-card/StatsCard.module.scss]:
```suggestion
    font-size: $scale-base;
```

File: src/stats-card/StatsCard.module.scss [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    font-weight: 600;
+    line-height: 15px;
```
Comment (richbachman) [inline src/stats-card/StatsCard.module.scss]:
```suggestion
    line-height: $scale-base-up-02;
```

File: src/stats-card/StatsCard.module.scss [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+    color: $slate;
+    text-transform: uppercase;
+  }
+
+  .stat {
+    padding-bottom: 7px;
+    font-size: 20px;
```
Comment (richbachman) [inline src/stats-card/StatsCard.module.scss]:
```suggestion
    font-size: $scale-base-up-04;
```

File: src/stats-card/StatsCard.module.scss [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+    font-weight: 600;
+    line-height: 30px;
```
Comment (richbachman) [inline src/stats-card/StatsCard.module.scss]:
```suggestion
    line-height: $scale-base-up-06;
```

File: src/stats-card/StatsCard.module.scss [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+  }
+
+  .stat,
+  .label {
+    margin-bottom: 0;
+    text-align: center;
+  }
+
+  &.secondary {
+    padding: 24px 30px;
```
Comment (richbachman) [inline src/stats-card/StatsCard.module.scss]:
```suggestion
    padding: 24px $scale-base-up-06;
```

File: src/stats-card/StatsCard.module.scss [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+
+    .stat {
+      font-size: 22px;
+      line-height: 20px;
```
Comment (richbachman) [inline src/stats-card/StatsCard.module.scss]:
```suggestion
      line-height: $scale-base-up-04;
```

File: src/stats-card/StatsCard.module.scss [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+    }
+
+    .stat.small-stat {
+      font-size: 15px;
```
Comment (richbachman) [inline src/stats-card/StatsCard.module.scss]:
```suggestion
      font-size: $scale-base-up-02;
```

**Comment by thinkingserious** [comment]
Hello @alexabreu,<br /><br/>Thanks again for the PR!<br />                                              <br />We appreciate your contribution and look forward to                                              continued collaboration. Thanks!                                             <br /><br />Team SendGrid DX

**Pull request merged**